packages/jellyfish-api-core/__tests__/category/spv/getHtlcSeed.test.ts

```diff
@@ -0,0 +1,78 @@
+    await container.start()
+    await container.call('spv_fundaddress', [await container.call('spv_getnewaddress')]) // Funds 1 BTC
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should getHtlcSeed', async () => {
+    const pubKeyA = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const pubKeyB = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const htlc = await container.call('spv_createhtlc', [pubKeyA, pubKeyB, '10'])
+
+    await container.call('spv_sendtoaddress', [htlc.address, 0.1]) // Funds HTLC address
+    await container.call('spv_claimhtlc', [htlc.address, await container.call('spv_getnewaddress'), htlc.seed]) // claim HTLC
+
+    const secret = await client.spv.getHtlcSeed(htlc.address)
+    expect(secret).toStrictEqual(htlc.seed)
+  })
+
+  it('should getHtlcSeed with empty secret before spv_claimhtlc', async () => {
+    const pubKeyA = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const pubKeyB = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const htlc = await container.call('spv_createhtlc', [pubKeyA, pubKeyB, '10'])
+
+    const secret = await client.spv.getHtlcSeed(htlc.address)
+    expect(secret).toStrictEqual('')
+  })
+
+  it('should not getHtlcSeed with invalid public key as receiverPubKey', async () => {
+    const promise = client.spv.getHtlcSeed('XXXX')
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow("RpcApiError: 'Error: Invalid address', code: -5, method: spv_gethtlcseed")
+  })
+})
+
+describe('Spv with ContainerGroup', () => {
+  const group = new ContainerGroup([
+    new MasterNodeRegTestContainer(GenesisKeys[0]),
+    new MasterNodeRegTestContainer(GenesisKeys[1])
+  ])
+  const client0 = new ContainerAdapterClient(group.get(0))
+  const client1 = new ContainerAdapterClient(group.get(1))
+
+  beforeAll(async () => {
+    await group.start()
+    await group.get(0).call('spv_fundaddress', [await group.get(0).call('spv_getnewaddress')]) // Funds 1 BTC
+  })
+
+  afterAll(async () => {
+    await group.stop()
+  })
+
+  it('should not get HTLC secret from another node', async () => {
+    const pubKeyA = await group.get(0).call('spv_getaddresspubkey', [await group.get(0).call('spv_getnewaddress')])
+    const pubKeyB = await group.get(0).call('spv_getaddresspubkey', [await group.get(0).call('spv_getnewaddress')])
+    const htlc = await group.get(0).call('spv_createhtlc', [pubKeyA, pubKeyB, '10'])
+
+    await group.get(0).call('spv_sendtoaddress', [htlc.address, 0.1]) // Funds HTLC address
+    await group.waitForSync()
+    await group.get(0).call('spv_claimhtlc', [htlc.address, await group.get(0).call('spv_getnewaddress'), htlc.seed]) // claim HTLC
+
+    const emptySecret = await client1.spv.getHtlcSeed(htlc.address)
```

Comment:
Hmm. This is incorrect as per my knowledge. once the secret is revealed, it is public. anyone can retrieve it.
consider the same scenario described here. other party, client1 should be able to retrieve the secret.

But i doubt that this `await group.waitForSync()` won't be enough for this. in real world, clients has to be synced in BTC network. means 10 mins for one block. Here in `regtest`, we don't even connect to BTC network. we just fake it. so a solution would be to do the faking in all nodes inside `group.waitForSync()`. But not sure it's an overkill. 
But the idea is anyone can retrive the secret once it's revealed.
@fuxingloh

Reply:
Ok, that's not what I observed while testing. As you pointed out, it must be because it is not synced to BTC network.
How could I test real multi node situation instead ?

Reply:
the answer is in my last comment. there are some fake/test spv RPCs that can be used to implement something like that. then it would be like we are faking and testing the same.

at this point, it does not make sense to use a multinode setup to test spv. even C++ side regtest tests are doing single node.

Reply:
Should we put on hold multi node testing on jellyfish side until C++ regtests do it ?

Reply:
Yeah, let's put multi-node testing on hold (`it.skip`). Since there isn't a proper way to peer connection in the BTC network.

Reply:
it.skip or just remove it ? As the logic is wrong anyway